### DISCUSSION TITLE
suggestion: set response status in `_return_json_response` with a different error key for the Json feature

### DISCUSSION THREAD

**Discussion by opya** [jeremyevans/rodauth#339] [General]
Hi all,

I'm trying to set the same key `:message` for both error and success keys `json_response_success_key` and `json_response_error_key` in order to get consistent results for example:

for failed login:

```json
{
  "errors": {
    "login": "no matching login"
  },
  "message": "There was an error logging in"
}
```
for successful login:
```json
{
  "message": "You have been logged in"
}
```

this is my setup

```ruby
plugin :rodauth, json: :only do
  enable  :login,
          :logout,
          :create_account,
          :change_password,
          :reset_password,
          :disallow_common_passwords,
          :jwt,
          :jwt_cors,
          :internal_request,
          :active_sessions,
          :audit_logging

  json_response_success_key     :message
  json_response_error_key       :message
  json_response_field_error_key :errors
end
```

But I keep getting a 400 status code because this check event if the login was successful https://github.com/jeremyevans/rodauth/blob/5c3747b40a1899064c0b4efe862dead72cfefc83/lib/rodauth/features/json.rb#L174-L178

is it wrong if we change it to check for `json_response[json_response_field_error_key]` or I'm doing something totally wrong

**Comment by jeremyevans** [comment]
I don't think we should change the default behavior.  If you are setting the same key for success and error, you are definitely going against how Rodauth is expected to be used.

To support what you want, we should extract a method named `json_response_error?`, have it do `!!json_response[json_response_error_key]` by default, and then you can override that method to get the behavior you want.

**Comment by opya** [comment]
if i understand correctly, we need a new method as you suggested

```ruby
def json_response_error?
  !!json_response[json_response_error_key]
end
```

and then we need to add it here as `auth_method` in order to overwrite it later
https://github.com/jeremyevans/rodauth/blob/5c3747b40a1899064c0b4efe862dead72cfefc83/lib/rodauth/features/json.rb#L23-L25

do you accept PR for this if in understanding it correctly

**Reply by jeremyevans** [reply]
Yep, that would be the general approach.
